test/fixtures/dialects/mysql/drop_function.yml

```diff
@@ -3,14 +3,14 @@
 # computed by SQLFluff when running the tests. Please run
 # `python test/generate_parse_fixture_yml.py`  to generate them after adding or
 # altering SQL files.
-_hash: ee089fc4f3feee06e8a9994f784339f57d053a3e80828d100506feb04e52ef23
+_hash: b887383bf035ac6c2e2e4357616d204661edd160b69fcecc0584f03a43ba1188
 file:
   statement:
-    drop_statement:
+    drop_function_ statement:
     - keyword: DROP
     - keyword: FUNCTION
     - keyword: IF
     - keyword: EXISTS
-    - table_reference:
+    - object_reference:
         identifier: '`testfunction`'
```

Comment:
wonder if a function reference exists that we can use in `mysql`s `drop_function_ statement` instead of `object_reference`

Reply:
addressed in my PR